playbooks/openstack/e2e.yaml

```diff
@@ -0,0 +1,186 @@
+          when: zuul_config is defined and zuul_config is not none and zuul_config != ''
+        - name: Print Zuul config  # TODO: Use the config and override `cluster_stack` dir if set
+          ansible.builtin.debug:
+            msg: "{{ zuul_config }}"
+    - name: Create cluster stack
+      ansible.builtin.command: "csctl create {{ project_dir }}/{{ cluster_stack }} --output {{ cluster_stack_release_dir }}  --mode hash"
+      args:
+        chdir: "{{ project_dir }}"
+      changed_when: true
+    - name: Ensure management cluster
+      block:
+        - name: Create management cluster config file
+          ansible.builtin.template:
+            src: "mgmt-cluster-config.yaml.j2"
+            dest: "{{ ansible_user_dir }}/mgmt-cluster-config.yaml"
+            mode: "0644"
+        - name: Create management cluster
+          ansible.builtin.command: "kind create cluster --config {{ ansible_user_dir }}/mgmt-cluster-config.yaml"
+          changed_when: true
+        - name: Wait for all system pods in the management cluster to become ready
+          ansible.builtin.command: "kubectl wait -n kube-system --for=condition=Ready --timeout={{ k8s_management_cluster_wait_for }}s pod --all"
+          changed_when: true
+    - name: Install CAPI and CAPO
+      ansible.builtin.command: "clusterctl init --infrastructure openstack:{{ capo_version }}"
+      changed_when: true
+      environment:
+        CLUSTER_TOPOLOGY: "true"
+        EXP_CLUSTER_RESOURCE_SET: "true"
+    - name: Install CSO and mount cluster stack release
+      ansible.builtin.import_tasks: ../tasks/cso.yaml
+      vars:
+        release_dir: "{{ cluster_stack_release_container_dir }}"
+    - name: Install CSPO and mount cluster stack release
+      ansible.builtin.import_tasks: ../tasks/cspo.yaml
+      vars:
+        release_dir: "{{ cluster_stack_release_container_dir }}"
+    - name: Read Zuul's clouds.yaml content, base64 encoded
+      ansible.builtin.slurp:
+        src: /etc/openstack/clouds.yaml
+      register: clouds_yaml_b64
+    - name: Read Zuul's secure.yaml content, base64 encoded
+      ansible.builtin.slurp:
+        src: /etc/openstack/secure.yaml
+      register: secure_yaml_b64
+    - name: Combine clouds_yaml_b64 and secure_yaml_b64 to produce full clouds.yaml
+      ansible.builtin.set_fact:
+        clouds_yaml_full: "{{ clouds_yaml_b64.content | b64decode | from_yaml | ansible.builtin.combine(secure_yaml_b64.content | b64decode | from_yaml, recursive=true) }}"
+    - name: Write clouds.yaml file
+      ansible.builtin.copy:
+        content: "{{ clouds_yaml_full | to_yaml }}"
+        dest: "{{ ansible_user_dir }}/clouds.yaml"
+        mode: "0644"
+    - name: Create secrets and ClusterResourceSet for the clusterstacks approach
+      ansible.builtin.shell:
+        cmd: |
+          set -o pipefail
+          helm template {{ openstack_csp_helper_chart_url }} -f {{ ansible_user_dir }}/clouds.yaml | kubectl apply -f -
```

Comment:
Is there a reason not to use a helm install? helm template omits several of helms features, for example capabilities, tests, notes, rollbacks, versioning of the provided values...

Reply:
Because when I started working on this e2e test, I followed the quickstart [guide](https://input.scs.community/_HeOTRCRSu2Uf2SfMSoOkQ).  I know that meanwhile it was changed in the `csp-helper` repo readme and it is recommended to do it via `helm install`. But basically, you do not need rollbacks, etc... in the E2E test pipeline. And as a reminder, this is just a `Draft`, which means it's WIP, so some things may be changed or added.

Reply:
Sure, sorry if i have been too quick with my critique. You should definitely have your time to build your PR before getting such feedback.

Somehow using helm template followed by kubectl apply became a thing, even if this renders a good amount of helms mechanisms useless. Even CNCF graduated projects (Argo) do it. I can see no reason to do it like that and a lot of reasons to use it the way it was designed. I try to counteract that and give my feedback early when i see this (again). That is the reason i reached out so early.

Reply:
It is ok, I already incorporated your comment.